src/AEG/CutNode.ts

```diff
@@ -115,32 +115,65 @@ export class CutNode {
     }
 
     /**
-     * Method that checks whether the given node can be inserted into this cut
-     * at a given point without overlapping any bounding boxes.
+     * Checks whether the incoming node can be inserted into this CutNode
+     * at a given point without overlapping any boundaries.
      * @param incomingNode The node to be inserted.
-     * @param insertionPoint The point at which the node must be inserted
      * @returns True, if the node can be inserted. Else, false
      */
-    public canInsert(incomingNode: AtomNode | CutNode, insertionPoint: Point): boolean {
-        //TO BE IMPLEMENTED??
-
-        const isValid = true;
-
+    public canInsert(incomingNode: AtomNode | CutNode): boolean {
+        let isValid = true;
+        let childRect = null;
+        let childEllipse = null;
+        let childEllipse2 = null; //for incomingNode as a CutNode and a CutNode child
+
+        if (incomingNode instanceof AtomNode && this.ellipse !== null) {
+            //INCOMING Atom on THIS Cut collision checking
+            isValid = (incomingNode as AtomNode).rect.overlaps(this.ellipse);
+            for (let i = 0; isValid && i < this.children.length; i++) {
```

Comment:
This is confusing control flow, why check isValid in the loop condition when you could just break if its false?